src/integrators/integrator_interface.jl

```diff
@@ -148,12 +148,15 @@ function resize_J_W!(cache, integrator, i)
     nf = nlsolve_f(f, integrator.alg)
     islin = f isa Union{ODEFunction,SplitFunction} && islinear(nf.f)
     if !islin
-      J = similar(f.jac_prototype, i, i)
-      if !isa(J, DiffEqBase.AbstractDiffEqLinearOperator)
+      if isa(cache.J, DiffEqBase.AbstractDiffEqLinearOperator)
+        resize!(cache.J,i)
```

Comment:
Is this right? Should it be `resize!(cache.J,i,i)`?

Reply:
It has to be square though, so I'm not sure if that's necessary? It would always just use the first value?

Reply:
Makes sense.